pkg/machine/qemu/machine.go

```diff
@@ -1735,3 +1735,11 @@ func isProcessAlive(pid int) bool {
 func (p *Provider) VMType() string {
 	return vmtype
 }
+
+func isRootfull() bool {
+	// Rootless is not relevant on Windows. In the future rootless.IsRootless
+	// could be switched to return true on Windows, and other codepaths migrated
+	// for now will check additionally for valid os.Getuid
+
+	return !rootless.IsRootless() && os.Getuid() != -1
```

Comment:
doesn't `rootless.IsRootless()` already return `false` on Windows?

Reply:
> doesn't rootless.IsRootless() already return false on Windows?

This is a problem. It returns false in the machine command, so, the code assumes that it is running in Rootful mode on windows and trying to put all .sock files in `/run`, which is irrelevant on Windows. Even if run with admin privileges I believe it is appropriate to user AppData for this purpose as it doesn't really change the user on Windows, but only elevates the rights.

So, `isRootless()` looks like always should return `true` on Windows.

So the logic is: it is rootful if not rootless and not windows. 
Negation (is rootless): it is rootless if rootless or windows.

It is not some reusable utility for other modules, but this logic makes sense for machine init command.

Reply:
thanks for the clarification